### DISCUSSION TITLE
Meaning of data from BLE message

### DISCUSSION THREAD

**Discussion by halinh12sinh** [alexanderlavrushko/BLE-HUD-navigation-ESP32#31] [Q&A]
Hi all,
Thank you for taking the time to read my question.
I did successful compile and deploy code to Lilygo T display.
I just want to know the meaning of the message from Sygic.
![image](https://github.com/alexanderlavrushko/BLE-HUD-navigation-ESP32/assets/52124894/f94bc0fe-03e4-4ada-b8ac-b3916d861cf2)
Example:
01: I don't know the meaning
32 Means speed limit (50kph)
0A: Turn indicator
30:  : I don't know the meaning
6D::  I don't know the meaning
Do you have any idea?

**Comment by alexanderlavrushko** [comment]
Hi @halinh12sinh,
Data meaning:
0x01 - type of the message (0x01 is "basic data" message, currently it's the only possible type)
0x32 - speed limit (50 km/h)
0x0A - "turn right" instruction
0x30 - ASCII symbol '0' (zero)
0x6D - ASCII symbol 'm'

The pair of 30 6D - is a text message "0m" meaning that instruction should be in zero meters, that is right now.
